### PR TITLE
Add Clangd language server support for C

### PR DESCRIPTION
**Pull request by melentye**
[Clangd](https://clang.llvm.org/extra/clangd.html) is a language server protocol implementation backed by Clang. I assume that from the linting perspective it should be fairly similar to what the `clang` linter provides already, however, being an LSP linter, `clangd` enables completion and go-to-definition.

The linter implementation follows examples for other LSP linters closely.

Haven't tried it with C++ but I assume brute copy-paste might just work.

### CODE REVIEW

File: test/command_callback/clangd_paths/dummy.c [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#include <stdio.h>
```
Comment (w0rp) [inline test/command_callback/clangd_paths/dummy.c]: Try deleting this file. The test won't read it.

**Comment by w0rp** [comment]
Cheers! :beers:

**Comment by melentye** [comment]
Thanks for the plugin and quick PR review! :beers:

**Comment by esran** [comment]
Should this support completion? Because I'm getting errors if I try and enable completion with this.

> Error detected while processing function
> ale#completion#OmniFunc[19]..ale#completion#ParseLSPCompletions:
> line   34:                                                                                                                                                                                                                                    
> E716: Key not present in Dictionary: kind is s:LSP_COMPLETION_METHOD_KIND
> Error detected while processing function
> ale#completion#OmniFunc[19]..ale#completion#ParseLSPCompletions:
> line   34:
> E15: Invalid expression: l:item.kind is s:LSP_COMPLETION_METHOD_KIND
> Pattern not found
> E117: Unknown function: lsp#complete
> E117: Unknown function: lsp#complete
> Pattern not found

**Comment by w0rp** [comment]
@esran I just pushed a commit to fix that. I wasn't handling the `kind` attribute being optional. https://github.com/w0rp/ale/commit/7bf3a749d088964b2ae42e8019dc6a570173d1bf

**Comment by esran** [comment]
Still getting some spurious messages sometimes. Looks like when completing members of structs.

> E117: Unknown function: lsp#complete
> E117: Unknown function: lsp#complete
> Pattern not found                                                                                                                                                                                                                             

Plus a lot of:

> Back at original

in the :messages log.

**Comment by w0rp** [comment]
The string `lsp#complete` doesn't exist in the ALE codebase. Try disabling other plugins.

**Comment by esran** [comment]
My bad. I'd disabled the plugins but not their config. All looks good now. Nice work :)

**Pull request merged**